### PR TITLE
Adding support for variable soil thickness to CTC

### PR DESCRIPTION
**Pull request by bishtgautam**
Adds consistency to run variable soil thickness with CTC, fixes a few
remaining bugs in variable soil thickness.

[BFB]

### CODE REVIEW

File: components/elm/src/biogeochem/PhosphorusDynamicsMod.F90 [hunk 1]
```diff
@@ -759,11 +759,41 @@ subroutine PhosphorusBiochemMin_balance(bounds,num_soilc, filter_soilc, &
         do j = 1,nlevdecomp
             biochem_pmin_to_ecosysp_vr_col(c,j)=0._r8
             do l = 1, ndecomp_pools
+<<<<<<< HEAD
```
Comment (peterdschwartz) [inline components/elm/src/biogeochem/PhosphorusDynamicsMod.F90:762]: Looks like a conflict from the rebase still needs to be resolved.

Comment (bishtgautam) [reply components/elm/src/biogeochem/PhosphorusDynamicsMod.F90:762]: Yes, my rebase messed this up. Let me clean this.

**Comment by bishtgautam** [comment]
@mabrunke @peterdschwartz I have fixed the rebase now.

**Review by mabrunke — APPROVED** [review]
Looks good to me.

**Comment by rljacob** [comment]
Status:  waiting on other PRs.

**Review by peterdschwartz — APPROVED** [review]

**Comment by peterdschwartz** [comment]
merged to next

**Comment by rljacob** [comment]
This new test ERS.f09_g16.I1850GSWCNPRDCTCBC.elm-vstrd is asking for a bunch of files to be downloaded.  Anyway to reduce that?
@jgfouca note above.  Mappy failed this test because it couldn't find a bunch of the new input.

**Comment by peterdschwartz** [comment]
@rljacob yeah I will push a fix or that soon

**Comment by peterdschwartz** [comment]
re-merged to next.

**Comment by peterdschwartz** [comment]
Modifying the test worked.  I have requested the DIFFs for the new test

**Comment by bishtgautam** [comment]
@mabrunke FYI, This has been merged to master.

**Pull request merged**